flipt-client-go/evaluation.go

```diff
@@ -228,35 +234,34 @@ func NewEvaluationClient(ctx context.Context, opts ...ClientOption) (_ *Evaluati
 		return
 	}
 
-	transport := &http.Transport{
-		DialContext: (&net.Dialer{
-			Timeout:   10 * time.Second, // connection timeout
-			KeepAlive: 30 * time.Second,
-		}).DialContext,
-	}
-
 	c.httpClient = &http.Client{
-		Transport: transport,
+		Transport: &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout:   10 * time.Second, // connection timeout
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+		},
 	}
 
-	switch c.fetchMode {
-	case FetchModeStreaming:
-		c.url = fmt.Sprintf("%s/internal/v1/evaluation/snapshots?[]namespaces=%s", c.url, c.namespace)
-
-	case FetchModePolling:
-		c.url = fmt.Sprintf("%s/internal/v1/evaluation/snapshot/namespace/%s", c.url, c.namespace)
-		if c.ref != "" {
-			c.url = fmt.Sprintf("%s?reference=%s", c.url, c.ref)
-		}
+	// Store the base URL without trailing slash
+	c.baseURL = strings.TrimRight(c.baseURL, "/")
 
-		// only set timeout for polling mode
-		// streaming mode will have no timeout set
+	// Set timeout only for polling mode
+	if c.fetchMode == FetchModePolling {
 		c.httpClient.Timeout = c.requestTimeout
+	}
 
-	default:
+	// Validate fetch mode
+	if c.fetchMode != FetchModePolling && c.fetchMode != FetchModeStreaming {
 		return nil, fmt.Errorf("invalid fetch mode: %s", c.fetchMode)
 	}
 
+	// Get initial snapshot URL (always uses polling endpoint)
+	initialURL := fmt.Sprintf(c.baseURL+snapshotPath, c.namespace)
+	if c.ref != "" {
+		initialURL += "?reference=" + c.ref
```

Comment:
this will fail for Flipt Cloud

Reply:
how so?

Reply:
The streaming base url has `?` . So adding extra one will break the url

Reply:
its only applied to `snapshotPath` which is `snapshotPath  = "/internal/v1/evaluation/snapshot/namespace/%s"`

Reply:
Oh... I missed that.